src/app/cake/cake.service.ts

```diff
@@ -0,0 +1,27 @@
+import { DataService } from "../../component/data.service";
+import { Cake } from "./data/cake.model";
+import { CakeSchema } from "./data/cake.schema";
+
+
+export class CakeService{
+
+    constructor(
+        private dataService: DataService<Cake> = new DataService('Cake', CakeSchema)
```

Comment:
Maybe moving the hard codded Cake to the model file as export const modelName: string = 'Cake', would be a better idea 😅